src/dialog/dlgreplacecuecolor.h

```diff
@@ -0,0 +1,69 @@
+#pragma once
+
+#include <QDialog>
+#include <QFuture>
+#include <QFutureWatcher>
+#include <QMap>
+#include <QMenu>
+#include <QProgressDialog>
+#include <QPushButton>
+#include <QSet>
+
+#include "dialog/ui_dlgreplacecuecolordlg.h"
+#include "library/dao/trackdao.h"
+#include "preferences/usersettings.h"
+#include "track/trackid.h"
+#include "util/color/rgbcolor.h"
+#include "util/db/dbconnectionpooled.h"
+#include "util/db/dbconnectionpooler.h"
+#include "widget/wcolorpickeraction.h"
+
+// Dialog for bulk replacing colors of cues in the Database.
+class DlgReplaceCueColor : public QDialog, public Ui::DlgReplaceCueColor {
+    Q_OBJECT
+  public:
+    enum class ConditionFlag {
+        NoConditions = 0,
+        CurrentColorCheck = 1,
+        CurrentColorNotEqual = 1 << 1,
+        HotcueIndexCheck = 1 << 2,
+        HotcueIndexNotEqual = 1 << 3,
+    };
+    Q_DECLARE_FLAGS(Conditions, ConditionFlag);
+
+    DlgReplaceCueColor(
+            UserSettingsPointer pConfig,
+            TrackDAO* pTrackDAO,
+            mixxx::DbConnectionPoolPtr dbConnectionPool,
+            QWidget* pParent);
+    ~DlgReplaceCueColor();
+
+  signals:
+    void databaseTracksChanged(QSet<TrackId> Trackids);
```

Comment:
We don't need to pass and connect TrackDAO. The dialog should only just a signal for the updated tracks, independent of who is connected.

Of course, using a compatible signature that allows us to directly connect this signal to a slot of TrackDAO is an obvious choice.